core/common/src/test/java/alluxio/ConfigurationTest.java

```diff
@@ -507,4 +507,20 @@ public void sitePropertiesLoadedNotInTest() throws Exception {
       assertEquals("TEST_LOGGER", Configuration.get(PropertyKey.LOGGER_TYPE));
     }
   }
+
+  @Test
+  public void discardIgnoredSitePropertiesTest() throws Exception {
+    Properties siteProps = new Properties();
+    siteProps.setProperty(PropertyKey.MASTER_HOSTNAME.toString(), "host-1");
+    siteProps.setProperty(PropertyKey.LOGS_DIR.toString(), "/tmp/logs1");
+    Map<String, String> sysProps = new HashMap<>();
+    sysProps.put(PropertyKey.LOGS_DIR.toString(), "/tmp/logs2");
+    try (Closeable p = new SystemPropertyRule(sysProps).toResource()) {
+      Configuration.discardIgnoredSiteProperties(siteProps);
+      Configuration.merge(siteProps);
+      Configuration.merge(sysProps);
```

Comment:
This hardcodes an assumption in the tests that `Configuration` will call these methods in a certain way. If `Configuration` changes in the future, e.g. someone accidentally removes the call to `Configuration.discardIgnoredSiteProperties(siteProps)`, this test won't catch that error. Is it possible to treat `Configuration` as a black box and verify that it doesn't accept ignored site properties from a site properties file?

Reply:
Done.